front/app/containers/Admin/formBuilder/containers/projects/forms/edit/index.tsx

```diff
@@ -37,69 +46,74 @@ const StyledRightColumn = styled(RightColumn)`
   overflow-y: auto;
 `;
 
-export const FormEdit = () => {
+export const FormEdit = ({ intl: { formatMessage } }: InjectedIntlProps) => {
   const [selectedField, setSelectedField] = useState<
-    IFlatCustomField | undefined
+    IFlatCustomFieldWithIndex | undefined
   >(undefined);
   const { projectId } = useParams() as { projectId: string };
 
-  const { formCustomFields, setFormCustomFields } = useFormCustomFields({
+  const { formCustomFields } = useFormCustomFields({
     projectId,
   });
 
+  const prevFormCustomFields = usePrevious(formCustomFields);
+
+  const schema = object().shape({
+    customFields: array().of(
+      object().shape({
+        title_multiloc: validateMultiloc(
+          formatMessage(messages.emptyTitleError)
+        ),
+        description_multiloc: object(),
+        required: boolean(),
+      })
+    ),
+  });
+
+  const methods = useForm({
+    mode: 'onBlur',
+    resolver: yupResolver(schema),
+  });
+
+  const { fields, append, remove, move } = useFieldArray({
+    name: 'customFields',
+    control: methods.control,
+  });
+
+  useEffect(() => {
```

Comment:
You can pass the custom field value directly to the form via `defaultValues` and not have this as an effect. You might have to fetch them in a parent component and pass them down as a prop, though.